spock-core/src/main/java/org/spockframework/runtime/model/FeatureInfo.java

```diff
@@ -98,6 +98,15 @@ public void setDataProcessorMethod(MethodInfo method) {
     return dataProviders;
   }
 
+  public DataProviderInfo getDataProvider(String variableName) {
+    for (DataProviderInfo dataProvider : dataProviders) {
+      if (dataProvider.getDataVariables().get(0).equals(variableName)){
```

Comment:
Could it be used also with data providers with more than one parameter (like `[foo, _, bar] << sql.execute(...)`)?